torchaudio/compliance/kaldi.py

```diff
@@ -0,0 +1,209 @@
+        dither (float): Dithering constant (0.0 means no dither). If you turn this off, you should set
+            the energy_floor option, e.g. to 1.0 or 0.1 (default = 1.0)
+        energy_floor (float): Floor on energy (absolute, not relative) in Spectrogram computation.  Caution:
+            this floor is applied to the zeroth component, representing the total signal energy.  The floor on the
+            individual spectrogram elements is fixed at std::numeric_limits<float>::epsilon(). (default = 0.0)
+        frame_length (float): Frame length in milliseconds (default = 25.0)
+        frame_shift (float): Frame shift in milliseconds (default = 10.0)
+        min_duration (float): Minimum duration of segments to process (in seconds). (default = 0.0)
+        preemphasis_coefficient (float): Coefficient for use in signal preemphasis (default = 0.97)
+        raw_energy (bool): If True, compute energy before preemphasis and windowing (default = True)
+        remove_dc_offset: Subtract mean from waveform on each frame (default = True)
+        round_to_power_of_two (bool): If True, round window size to power of two by zero-padding input
+            to FFT. (default = True)
+        sample_frequency (float): Waveform data sample frequency (must match the waveform file, if
+            specified there) (default = 16000.0)
+        snip_edges (bool): If True, end effects will be handled by outputting only frames that completely fit
+            in the file, and the number of frames depends on the frame_length.  If False, the number of frames
+            depends only on the frame_shift, and we reflect the data at the ends. (default = true)
+        subtract_mean (bool): Subtract mean of each feature file [CMS]; not recommended to do
+            it this way.  (default = False)
+        window_type (str): Type of window ('hamming'|'hanning'|'povey'|'rectangular'|'blackman') (default = 'povey')
+
+    Not used Inputs:
+        allow_downsample (bool): If True, allow the input waveform to have a higher frequency than the specified
+            `sample_frequency` (and we'll downsample). (default = False)
+        allow_upsample (bool): If True, allow the input waveform to have a lower frequency than the specified
+            `sample_frequency` (and we'll upsample). (default = False)
+        max_feature_vectors (int): Memory optimization. If larger than 0, periodically remove feature vectors so
+            that only this number of the latest feature vectors is retained. (default = -1)
+        output_format (str): Format of the output files [kaldi, htk] (default = 'kaldi')
+
+    Outputs:
+        Tensor: a spectrogram identical to what Kaldi would output. The shape is (, `padded_window_size` // 2 + 1)
+    """
+    # numeric_limits<float>::epsilon()
+    epsilon = torch.tensor(1.19209290e-07, dtype=torch.get_default_dtype())
+
+    waveform = sig[max(channel, 0), :]  # size (n)
+    window_shift = int(sample_frequency * 0.001 * frame_shift)
```

Comment:
nit: create named variables for constants like this to improve readability.

Reply:
Done.